chapter4.adoc

```diff
@@ -2,17 +2,17 @@
 [[Smmtt]]
 == `Smmtt`: Hart MTT Extension
 
-=== Smmtt[34, 46, 56]rw
+[[Smmtt-rw]]
+=== Smmtt[34 | 46 | 56]rw
 
 The `PPN` rooted structure for the MTT is shown below. The structure
 below shows a 56 bit physical address lookup; for lower physical address
-widths e.g. 46 bits, the `MTTL3` table is not applicable. In this mode of the
-Smmtt[34, 46, 56]rw, each page is associated with a read and a write access
-permission (2 bits). <<mtt-map-rw>> shows the Physical address (PA) being
-used to index into the `MTT` structure in memory to lookup access
-permissions for the supervisor domain specified in the `MTTL1` entry.
-Intermediate `MTTL3` and `MTTL2` entries are used to allow this structure to
-be sparsely populated.
+widths e.g. 46 bits, the `MTTL3` table is not applicable. Each
```

Comment:
Could we add a note that the sizes shown for L3 and L2 tables are their maximum size. However only the parts of the table that correspond to memory present in the platform need to be actually available. Could give an example of where if a machine only has 40 bit physical memory as an example to show that only first N entries of L2 are accessed.

Reply:
Having the SPA being show as having 4 pn parts may want to be a separate diagram and it can be referenced from the walk section. The table layout can then be another diagram (which also includes the PA with pn split). Alternately some way to draw attention that "pn[i]" in walk process actually refers to what is shown in this diagram would help.

Reply:
updated to show SPA separately also